tests/integration-service/integration-with-env.go

```diff
@@ -0,0 +1,172 @@
+package integration
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/konflux-ci/e2e-tests/pkg/clients/has"
+	"github.com/konflux-ci/e2e-tests/pkg/framework"
+	"github.com/konflux-ci/e2e-tests/pkg/utils"
+	pipeline "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	integrationv1beta1 "github.com/konflux-ci/integration-service/api/v1beta1"
+	intgteststat "github.com/konflux-ci/integration-service/pkg/integrationteststatus"
+
+	appstudioApi "github.com/konflux-ci/application-api/api/v1alpha1"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = framework.IntegrationServiceSuiteDescribe("Integration Service E2E tests", Label("integration-service", "HACBS", "integration-env"), func() {
+	defer GinkgoRecover()
+
+	var f *framework.Framework
+	var err error
+
+	var applicationName, componentName, testNamespace string
+	var integrationTestScenario *integrationv1beta1.IntegrationTestScenario
+	var timeout, interval time.Duration
+	var originalComponent *appstudioApi.Component
+	var pipelineRun, integrationPipelineRun *pipeline.PipelineRun
+	var snapshot *appstudioApi.Snapshot
+
+	AfterEach(framework.ReportFailure(&f))
+
+	Describe("with happy path for general flow of Integration service with ephemeral environment", Ordered, func() {
+		BeforeAll(func() {
+			// Initialize the tests controllers
+			f, err = framework.NewFramework(utils.GetGeneratedNamespace("integration-env"))
+			Expect(err).NotTo(HaveOccurred())
+			testNamespace = f.UserNamespace
+
+			applicationName = createApp(*f, testNamespace)
+			componentName, originalComponent = createComponent(*f, testNamespace, applicationName)
+			integrationTestScenario, err = f.AsKubeAdmin.IntegrationController.CreateIntegrationTestScenario("", applicationName, testNamespace, gitURL, "main", pathIntegrationPipelineWithEnv)
```

Comment:
would it make sense to update `revision` const value and use it here instead of "main"?

Reply:
used const `itsRevision` , wdyt?

Reply:
hmm I meant to use the updated `revision` variable, because referring to branches that are getting updated independently on e2e-tests is not safe.

Reply:
got it, 
yes that would be safe one, 
I've updated the revision to point to the updated revision and it pass on my local cluster